src/js/jsx/search/SearchBar.jsx

```diff
@@ -400,6 +400,19 @@ define(function (require, exports, module) {
         /** @ignore */
         _handleDialogClick: function (event) {
             this.refs.datalist.removeAutofillSuggestion();
+            
+            // If string, it means no items match user inputted search query
+            var filters = this.state.filter,
+                lastFilter = filters[filters.length - 1];
+                
+            var flux = this.getFlux(),
+                searchStore = flux.store("search"),
+                payload = searchStore.getPayloadFromFilter(lastFilter),
+                noOptionsStock = nls.localize("strings.SEARCH.NO_OPTIONS_STOCK");
+
+            if (payload && payload.noOptions().noOptionsString === noOptionsStock) {
```

Comment:
You've created an explicit dependency on Stock. There should be no explicit dependency on any provider. Consider that if this ever gets moved to standard PS we will have to shuffle the set of providers somewhat, and we can't predict how exactly. So, there should be no explicit dependencies in any generic infrastructure on specific implementations.